web/src/localization/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { mapBundleSync } from '@fluent/sequence';
+import { negotiateLanguages } from '@fluent/langneg';
+import { Accessor, JSX, createContext, createMemo, useContext } from 'solid-js';
+import { createLocalStorageSignal } from '../utils/localstorage';
+import { Option } from '../types';
+
+const ftls = import.meta.glob('./locale/*.ftl', { eager: true, as: 'raw' });
+const bundles = Object.fromEntries(
+  Object.entries(ftls).map(([fileName, ftl]) => {
+    const languageName = fileName.replace(
+      /^.*\/(.+)\..*$/,
+      (_, language: string) => language,
+    );
+    const resource = new FluentResource(ftl);
+    const bundle = new FluentBundle(languageName);
+    const errors = bundle.addResource(resource);
+    if (errors.length > 0) {
+      throw new Error(
+        `Failed to load translation bundle ${fileName} as ${languageName}`,
+        { cause: errors[0] },
+      );
+    }
+    return [languageName, bundle];
+  }),
+);
+
+const fallbackLanguage = 'en';
```

Comment:
We might want to use uppercase here.

Reply:
Hm?

Reply:
I think it makes sense to use `FALLBACK_LANGUAGE` instead of `fallbackLanguage` for the variable name.

Reply:
21837f6b748f837f162bf7c3ffcdaa482e5ebfcd

Reply:
Great choice of a commit title :)